src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/RunnableProjectGenerator.cs

```diff
@@ -363,6 +362,24 @@ internal static object DataTypeSpecifiedConvertLiteral(IEngineEnvironmentSetting
             }
         }
 
+        private static bool TryResolveChoiceValue(string literal, ITemplateParameter param, out string match)
+        {
+            if (literal == null)
+            {
+                match = null;
+                return false;
+            }
+
+            match = param.Choices.Keys.FirstOrDefault(x => string.Equals(x, literal, StringComparison.OrdinalIgnoreCase));
+
+            if (match == null && param.Choices.Keys.Count(x => x.StartsWith(literal, StringComparison.OrdinalIgnoreCase)) == 1)
```

Comment:
An extension like won't quite work here because the exact match (line 373) short circuits the starts with match. So everything must be checked. But I'll re-write this to only iterate once.